tests/unit/test_cloud_integration.py

```diff
@@ -0,0 +1,84 @@
+import tempfile
+import unittest
+from pathlib import Path
+from unittest.mock import patch
+
+import yaml
+from charm import GrafanaAgentMachineCharm as GrafanaAgentCharm
+from ops import ActiveStatus, BlockedStatus
+from ops.testing import Harness
+
+
+class TestUpdateStatus(unittest.TestCase):
```

Comment:
I wonder how much of this test should reside in `canonical/grafana-cloud-integrator`.

Reply:
I think the only part that's testable in grafana-cloud-integrator is added already [here](https://github.com/canonical/grafana-cloud-integrator/pull/13).  The rest of this test is grafana-agent business logic (confirm the settings appear in the place they need to be).  Although tbh this unit test is a pretty large unit test, more an integration test that just doesn't need juju...